src/github.com/matrix-org/dendrite/common/events.go

```diff
@@ -38,25 +40,44 @@ var ErrRoomNoExists = errors.New("Room does not exist")
 // the room doesn't exist
 // Returns an error if something else went wrong
 func BuildEvent(
-	ctx context.Context,
+	req *http.Request,
```

Comment:
I am sorry I could not review this before.
I feel it's not common's job to parse the http request. Since this is used from all components, it will be very messy to send in the http.Request whenever we want a event to be built.
e.g. If someone wants to build an event by reading from kafka logs.

It'd be a lot cleaner if we could just pass the timestamp as an argument to BuildEvent.